src/rules/pydocstyle/rules/no_surrounding_whitespace.rs

```diff
@@ -32,7 +32,9 @@ pub fn no_surrounding_whitespace(checker: &mut Checker, docstring: &Docstring) {
         if let Some(pattern) = leading_quote(contents) {
             // If removing whitespace would lead to an invalid string of quote
             // characters, avoid applying the fix.
-            if !trimmed.ends_with(pattern.chars().last().unwrap()) {
+            if !trimmed.ends_with(pattern.chars().last().unwrap())
+                && !trimmed.starts_with(pattern.chars().next().unwrap())
```

Comment:
Though it’s not intuitive, I think this should also use last and next. Last will always be the quote char, but next could be a kind prefix (like r, if this is a raw string). Same in the other file.

Reply:
oh right, but why last and next? I would say only last. If next() is `r` it would ignore a string starting with `r`..

Reply:
Sorry, typo — should read “last, not next”.

Reply:
As in, we shouldn’t use “next” here, just “last” for both condition checks.

Reply:
done